edb/edgeql/parser/parser.py

```diff
@@ -36,9 +37,64 @@ def get_exception(self, native_err, context, token=None):
         else:
             if msg.startswith('Unexpected token: '):
                 token = token or getattr(native_err, 'token', None)
+                ltok = self.parser._stack[-1][0]
 
                 if not token or token.kind() == 'EOF':
                     msg = 'Unexpected end of line'
+                elif (
+                    token.kind() == 'IDENT' and
+                    isinstance(ltok, parsing.Nonterm)
+                ):
+                    # Make sure that the previous element in the stack
+                    # is some kind of Nonterminal, because if it's
+                    # not, this is probably not an issue of a missing
+                    # COMMA.
+
+                    msg = f'Unexpected {token.text()!r}'
+                    # There are a few things this could be, so we make
+                    # an attempt to distinguish some specific parsing
+                    # error in shapes.
+
+                    # Then check if LBRACE, LPAREN, or LBRACKET is in
+                    # the parser stack. Any one of those indicates
+                    # that the parser was in the middle of processing
+                    # a list of some sort (shape, tuple, or array) and
+                    # encountering an unexpected IDENT probably
+                    # indicates a missing COMMA instead.
+                    for i, (el, _) in enumerate(reversed(self.parser._stack)):
```

Comment:
Why do we need a `for` here? Looks like we break in almost every branch. Also in the first iteration of this loop, `el` would be the same as `ltok`, right? So maybe there's no need in looping here?